ktor-server/ktor-server-plugins/ktor-server-websockets/jvm/test/io/ktor/tests/websocket/WebSocketTest.kt

```diff
@@ -118,593 +106,402 @@ class WebSocketTest {
+        routing {
+            webSocket("/echo") {
+                assertFailsWith<ClosedReceiveChannelException> {
+                    receiveDeserialized<Data>()
                 }
+                executed.complete()
             }
-            val sendBuffer = ByteBuffer.allocate(100)
-
-            Serializer().apply {
-                enqueue(Frame.Text(""))
-                serialize(sendBuffer)
-            }
+        }
 
-            val conversation = Job()
-
-            handleWebSocket("/echo") {
-                bodyChannel = writer {
-                    channel.writeFully(sendBuffer.array())
-                    channel.flush()
-                    conversation.join()
-                }.channel
-            }.let {
-                runBlocking {
-                    withTimeout(Duration.ofSeconds(10).toMillis()) {}
-                }
-            }
+        createWebSocketsClient().webSocket("/echo") {
+            close()
         }
+
+        executed.complete()
     }
 
     @Test
-    fun testFrameSize() {
-        withTestApplication {
-            application.install(WebSockets)
-
-            application.routing {
-                webSocketRaw("/echo") {
-                    outgoing.send(Frame.Text("+".repeat(0xc123)))
-                    outgoing.send(Frame.Close())
-                }
-                webSocketRaw("/receiveSize") {
-                    val frame = incoming.receive()
-                    val bytes = buildPacket {
-                        writeInt(frame.buffer.remaining())
-                    }
+    fun testFrameSize() = testApplication {
+        install(WebSockets)
 
-                    outgoing.send(Frame.Binary(true, bytes))
-                    outgoing.send(Frame.Close())
-                }
+        routing {
+            webSocket("/echo") {
+                send("+".repeat(0xc123))
             }
 
-            handleWebSocket("/echo") {
-                setBody(byteArrayOf())
-            }.let { call ->
-                assertEquals("817ec123", hex(call.response.byteContent!!.take(4).toByteArray()))
-                call.response.awaitWebSocket(Duration.ofSeconds(10))
-            }
+            webSocket("/receiveSize") {
+                val frame = incoming.receive()
+                val bytes = buildPacket { writeInt(frame.data.size) }.readByteArray()
 
-            handleWebSocket("/receiveSize") {
-                setBody(hex("0x81 0x7e 0xcd 0xef".trimHex()) + "+".repeat(0xcdef).toByteArray())
-            }.let { call ->
-                assertEquals("82040000cdef", hex(call.response.byteContent!!.take(6).toByteArray()))
-                call.response.awaitWebSocket(Duration.ofSeconds(10))
+                send(bytes)
             }
         }
-    }
-
-    @Test
-    fun testMasking() {
-        withTestApplication {
-            application.install(WebSockets)
-            application.routing {
-                webSocketRaw("/echo") {
-                    masking = true
```

Comment:
I've removed this test. It seems like masking [was enabled by default](https://github.com/ktorio/ktor/issues/423), and I have no idea how to test it without access to the actual byte stream.